cmd/cosign/cli/sign/sign.go

```diff
@@ -244,36 +210,33 @@ func signDigest(ctx context.Context, digest name.Digest, payload []byte, ko KeyO
 		}
 		defer out.Close()
 	}
-	if _, err := out.Write([]byte(b64sig)); err != nil {
-		return errors.Wrap(err, "write signature to file")
-	}
 
-	if !upload {
-		return nil
+	var s icos.Signer
+	s = ipayload.NewSigner(sv, nil, nil)
+	s = ifulcio.NewSigner(s, sv.Cert, sv.Chain)
```

Comment:
A few things about this line don't sit right with me:
1. Shouldn't it be predicated on some condition?
2. I still don't like the fact that the fulcio Signer wraps another random Signer where we have zero visibility into what it does.  I think the reality here is that the payload Signer is doing the fulcio signing, and the fulcio Signer is just decorating the resulting oci.Signature to match.  I think this is a fragile composition, and I think I'd rather see payload's Signer be a hidden implementation detail of a fulcio Signer than allow arbitrary composition here.

Reply:
yes :p

"TODO:"
"* move over fulcio operations as well"